### PR TITLE
Closes #6: Deploy to dev server

### LINKED ISSUE #6: Deploy to dev server
**Describe the bug**
Following steps need to be implemented:
- Push the artifact to dev server.
- Push ṭhe systemd file to dev server.
- Start the servIce.

### PR DESCRIPTION
**Pull request by aprajshekhar**
Fixes #6 

## Proposed Changes

  - Adds build workflow
  - Adds deploy workflow
  -

### CODE REVIEW

File: .github/workflows/build_and_deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+      - main
+      - feat/ci-cd
```
Comment (picksitquick) [inline .github/workflows/build_and_deploy.yaml:7]: Is this branch still needed? this tells that only build when changes are made either on main branch or feat/ci-cd.

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ # copy the new JAR file to the remote directory and rename it
+
+sshpass -p $SSH_PASS  scp -v -o StrictHostKeyChecking=no  $JAR_FILE "$SSH_USERNAME@$SSH_HOST:$REMOTE_DIRECTORY/$REMOTE_JAR_FILE"
+echo "=========================="
+echo "==Starting service======="
+sshpass -p $SSH_PASS ssh "$SSH_USERNAME@$SSH_HOST" 'chown -R gamedoora:gamedoora /opt/gamedoora; chmod ug+rwx /opt/gamedoora/*.jar; systemctl restart gamedoora-gateway; systemctl is-active --quiet gamedoora-gateway && echo Service is running'
+echo "========================="
+# # Check the exit status of the previous command and set an output variable accordingly
+
+if [ $? -eq 0 ]; then
```
Comment (picksitquick) [inline deploy.sh:44]: do we need to change the code to 143 or keep it as 0 ?

**Review by picksitquick — COMMENTED** [review]
LGTM, just 2 comments please take a look.

File: .github/workflows/build_and_deploy.yaml (hunk 1 above)
Comment (aprajshekhar) [reply .github/workflows/build_and_deploy.yaml:7]: This is for future edge case implementation

File: deploy.sh (hunk 2 above)
Comment (aprajshekhar) [reply deploy.sh:44]: The exit code is not being passed back from ssh command. When we figure out a way to do that, the status code of the service can be used. Until then, my idea is to keep the current check.

**Review by picksitquick — APPROVED** [review]
Sounds good!!

**Pull request merged**